docs/guides/testing.mdx

```diff
@@ -12,7 +12,7 @@ Thus, there can be cases where the state of one test can affect another. To make
 
 ```jsx
 import { create as actualCreate } from 'zustand'
-// const actualCreate = jest.requireActual('zustand') // if using jest
+// const { create: actualCreate } = jest.requireActual('zustand') // if using jest
```

Comment:
Why `jest.requireActual`? Won't the line above
(`import { create as actualCreate } from 'zustand'`)
be sufficient even with `jest`?

From what I understood from [the Jest docs](https://jestjs.io/docs/jest-object#jestrequireactualmodulename),
`jest.requireActual` is for bypassing the `jest.mock`,
when we want to get to the actual implementation
from inside the mock.
Here, there is no `jest.mock('zustand')`,
so no way for Jest to confuse the implementations.

I see that it wasn't your code before,
but I wanted to drop in when we are at it :)

Reply:
@sewera I've fixed the docs to use plain `import` statements. I also fixed some mistakes in the Typescript docmentation. The `create` function was curried there for no reason, and the file type was TSX, which it didn't need to be. 

(The deprecation warnings I was getting were due to me using the wrong type before and are no longer a problem.)

Reply:
Hello,
First, thanks for picking these fixes up. I was going to open a PR but I realised you were already working on it.

I believe the need to use the `jest.requireActual` is to prevent circular dependencies.
If you write a test for a store you've created, the mock is going to run and it will import `zustand` which is the mock which will import the zustand mock again and it goes like this forever. See the screenshot:
![Screenshot 2023-02-23 at 17 00 41](https://user-images.githubusercontent.com/3709948/220962015-080376b4-e3fe-47ad-9771-124f1ee11d7f.png).

So, I would suggest to keep the commented line, but I would make it type safe by doing the following:
```ts
const actualCreate = jest.requireActual<typeof import("zustand")>("zustand").create; // if using jest
```

Reply:
That's a great suggestion, thanks!

It's curious, I'm not getting that problem in our project. But [madge](https://www.npmjs.com/package/madge) flags it as a circular dependency.

I guess we'll keep it with `jest.useActual`—your suggestion removes the only drawback. I'll update the PR.